File: solutions/swift/triangle/1/Sources/Triangle/Triangle.swift
struct Triangle {
    let a, b, c: Double

    init(_ a: Double, _ b: Double, _ c: Double) {
        self.a = a
        self.b = b
        self.c = c
    }

    var kind: String {
        guard (a + b > c) && (b + c > a) else {
            return "ErrorKind"
        }

        switch true {
        case a == b && b == c:
            return "Equilateral"
        case a == b, c == b, c == a:
            return "Isosceles"
        default:
            return "Scalene"
        }
    }
}

File: solutions/go/triangle/2/triangle.go
package triangle

import "math"

type Kind int

const (
	NaT = iota // not a triangle
	Equ        // equilateral
	Iso        // isosceles
	Sca        // scalene
)

// KindFromSides return triangle type from it's side
func KindFromSides(a, b, c float64) Kind {
	for _, side := range []float64{a, b, c} {
		if math.IsNaN(side) || math.IsInf(side, 1) || math.IsInf(side, -1) || side == 0 {
			return NaT
		}
	}
	if a+b < c || a+c < b || b+c < a {
		return NaT
	}
	if a == b && a == c {
		return Equ
	}
	if a == b || a == c || b == c {
		return Iso
	}
	return Sca
}
